catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/RoleResourceTest.java

```diff
@@ -115,7 +115,7 @@ public Role createRolesAndSetDefault(TestInfo test, @Positive int numberOfRoles,
       throws IOException {
     // Create a set of roles.
     for (int i = 0; i < numberOfRoles; i++) {
-      CreateRole create = createRequest(test, offset + i + 1);
```

Comment:
I think this is a bug that caused some tests to fail. 

If offset is 10 and numberOfRoles is 7, the range would be [11,17]. (offset + i + 1)

Below on line 125, we pick offset + random.nextInt(numberOfRoles), which is within range [10,16]

Since the range is not the same this could possibly fail.
For ex, if line 125 picks 10, it would not find the role because the roles are numbered 11-17. 

So this should fix #2959
Will try rerunning tests a few times to confirm


cc: @vivekratnavel

Reply:
Confirmed that this test did not break in the next 3 runs

Reply:
Great. Thank you!